src/wmtk/operations/tri_mesh/VertexSmoothUsingDifferentiableEnergy.cpp

```diff
@@ -10,6 +11,9 @@ void OperationSettings<tri_mesh::VertexSmoothUsingDifferentiableEnergy>::initial
     base_settings.initialize_invariants(m);
     base_settings.invariants.add(
         std::make_shared<TriangleInversionInvariant>(m, coordinate_handle));
+    if (!smooth_boundary) {
+        base_settings.invariants.add(std::make_unique<InteriorVertexInvariant>(m));
```

Comment:
That sounds to me like it should not be done by having an operation setting but actually by adding an invariant on the user side. If what you did here was for the purpose of backwards compatibility, please write a comment so that we can refactor it in the future.

Reply:
Isn't this a property in the base settings? We should just use that one's settings